### ISSUE TITLE
contacts and distributions as array?

### ISSUE THREAD

**Issue by pvgenuchten** [geopython/pygeometa#219]
The pattern of having contacts and distributions as key-value pairs is common in mcf. However this pattern is problematic in [mdme](https://github.com/osgeo/mdme). the `build form from json schema` is not able to manage this pattern in json-schema:

```
"contact": {
            "title": "Contacts",
            "patternProperties": {
                "^.*": {
                    "type": "object",
``` 

I wonder if there are other scenario's with the same issue (there is for example the [PR with multiple contacts having the same role](#196) which is used as unique key).

I understand this is a massive breaking change, but just curious to hear your thoughts.

**Comment by pvgenuchten** [comment]
in this [pr](https://github.com/OSGeo/mdme/pull/22) i convert from object to array at load and back to object at save.... it works, but less optimal

**Comment by nicokant** [comment]
I agree that it should be possible to use arrays inside roles instead of just objects, some metadata formats allow for example to have multiple authors or creators. Would it be a problem to accept both an object and a list of objects?
